src/test/test_util_tests.cpp

```diff
@@ -0,0 +1,308 @@
+    BOOST_TEST(SCRIPT_VERIFY_TAPROOT == ParseScriptFlags("TAPROOT"sv, false));
+    BOOST_TEST((SCRIPT_VERIFY_MINIMALIF | SCRIPT_VERIFY_WITNESS) == ParseScriptFlags("MINIMALIF,WITNESS"sv, false));
+    BOOST_TEST((SCRIPT_VERIFY_TAPROOT | SCRIPT_VERIFY_LOW_S | SCRIPT_VERIFY_DERSIG) == ParseScriptFlags("TAPROOT,LOW_S,DERSIG"sv, false));
+    BOOST_TEST(0 == ParseScriptFlags("F00BAR"sv, false));
+    BOOST_TEST(0 == ParseScriptFlags("witness"sv, false));        // case-sensitive
+    BOOST_TEST(0 == ParseScriptFlags("P2SH , WITNESS"sv, false)); // Underlying function "split" doesn't trim
+
+    BOOST_TEST("STRICTENC"sv == FormatScriptFlags(SCRIPT_VERIFY_STRICTENC));
+    BOOST_TEST("WITNESS_PUBKEYTYPE"sv == FormatScriptFlags(SCRIPT_VERIFY_WITNESS_PUBKEYTYPE));
+    BOOST_TEST("CONST_SCRIPTCODE,TAPROOT"sv == FormatScriptFlags(SCRIPT_VERIFY_TAPROOT | SCRIPT_VERIFY_CONST_SCRIPTCODE));
+    BOOST_TEST("CLEANSTACK,NULLDUMMY,NULLFAIL"sv == FormatScriptFlags(SCRIPT_VERIFY_NULLDUMMY | SCRIPT_VERIFY_CLEANSTACK | SCRIPT_VERIFY_NULLFAIL));
+    BOOST_TEST(""sv == FormatScriptFlags(0));
+    BOOST_TEST(""sv == FormatScriptFlags(1U << 30)); // invalid flags just ignored
+    BOOST_TEST("WITNESS"sv == FormatScriptFlags((1U << 30) | SCRIPT_VERIFY_WITNESS));
+
+    // Systematic
+    auto flag_map = MapFlagNames();
+    for (auto [name1, value1] : flag_map) {
+        BOOST_TEST(value1 == ParseScriptFlags(name1));
+        BOOST_TEST(name1 == FormatScriptFlags(value1));
+
+        for (auto [name2, value2] : flag_map) {
+            if (value1 == value2) continue;
+            BOOST_TEST((value1 | value2) == ParseScriptFlags(std::string(name1) + "," + std::string(name2), false));
+            BOOST_TEST(std::string(name1 < name2 ? name1 : name2) + "," + std::string(name1 < name2 ? name2 : name1) == FormatScriptFlags(value1 | value2));
+        }
+    }
+}
+
+BOOST_AUTO_TEST_CASE(pretty_script_errors)
+{
+    // A small assortment
+    BOOST_TEST("OK"sv == FormatScriptError(SCRIPT_ERR_OK, false));
+    BOOST_TEST("UNKNOWN_ERROR"sv == FormatScriptError(SCRIPT_ERR_UNKNOWN_ERROR, false));
+    BOOST_TEST(""sv == FormatScriptError(SCRIPT_ERR_ERROR_COUNT, false));
+
+    BOOST_TEST(SCRIPT_ERR_OK == ParseScriptError("OK"sv, false));
+    BOOST_TEST(SCRIPT_ERR_SCHNORR_SIG_SIZE == ParseScriptError("SCHNORR_SIG_SIZE"sv, false));
+    BOOST_TEST(SCRIPT_ERR_UNKNOWN_ERROR == ParseScriptError("F00Bar"sv, false));
+    BOOST_TEST(SCRIPT_ERR_UNKNOWN_ERROR == ParseScriptError("Schnorr_Sig"sv, false)); // case-sensitive
+    BOOST_TEST(SCRIPT_ERR_UNKNOWN_ERROR == ParseScriptError("MINIMALIF "sv, false));  // doesn't trim arg
+
+    // Systematic
+    for (size_t i = 0; i < SCRIPT_ERR_ERROR_COUNT; ++i) {
+        const auto name = FormatScriptError(static_cast<ScriptError_t>(i));
+        BOOST_TEST(!name.empty());
+        BOOST_TEST(i == ParseScriptError(name));
+    }
+}
+
+#define EVAL(...) []() -> bool { return __VA_ARGS__; }()
+
+BOOST_AUTO_TEST_CASE(is_base_of_trait)
```

Comment:
https://github.com/bitcoin/bitcoin/pull/25097/commits/12ae33708ee1978e2a215954e41e6ccf4f0a25fd

Is there a reason to include this test case? This doesn't look like it's testing any non-std functionality. It also seems like a direct adaptation of the example usage here: https://en.cppreference.com/w/cpp/types/is_base_of